CSharpExtractor/CSharpExtractor/Extractor/Extractor.cs

```diff
@@ -104,29 +108,32 @@ private string PathToString(PathFinder.Path path)
 			return builder.ToString();
 		}
 
-		internal IEnumerable<PathFinder.Path> GetInternalPaths(Tree tree)
-		{
+        internal IEnumerable<PathFinder.Path> GetInternalPaths(Tree tree)
+        {
             var finder = new PathFinder(tree, LengthLimit, WidthLimit);
-            foreach (Tuple<Variable, Variable> varPair in 
-			         Utilities.WeakConcat(Utilities.Choose2(variables), 
-			                              variables.Select((arg) => new Tuple<Variable,Variable>(arg,arg))))
-			{
-				bool pathToSelf = varPair.Item1 == varPair.Item2;
-
-				foreach(var lhs in varPair.Item1.Leaves)
-				foreach (var rhs in varPair.Item2.Leaves)
-				{
-					if (lhs == rhs)
-							continue;
-					
-					PathFinder.Path path = finder.FindPath(lhs, rhs, limited: true);
-
-					if (path == null)
-						continue;
-                    
-
-					yield return path;
-				}
+
+            var all_the_pairs = Utilities.ReservoirSample(Utilities.WeakConcat(Utilities.Choose2(variables),
```

Comment:
can you please keep the consistency with naming conventions?
maybe: `allPairs`

Reply:
Done